pkg/scheduler/framework/plugins/nodeports/node_ports_test.go

```diff
@@ -163,9 +175,13 @@ func TestPreFilterDisabled(t *testing.T) {
 	p, _ := New(nil, nil)
 	cycleState := framework.NewCycleState()
 	gotStatus := p.(framework.FilterPlugin).Filter(context.Background(), cycleState, pod, nodeInfo)
-	wantStatus := framework.AsStatus(fmt.Errorf(`reading "PreFilterNodePorts" from cycleState: %w`, fmt.Errorf("not found")))
-	if !reflect.DeepEqual(gotStatus, wantStatus) {
-		t.Errorf("status does not match: %v, want: %v", gotStatus, wantStatus)
+	wantStatus := framework.AsStatus(fmt.Errorf(`reading "PreFilterNodePorts" from cycleState: %w`, errors.New("not found")))
+	cmpopt := cmp.Comparer(func(s1, s2 *framework.Status) bool {
+		return reflect.DeepEqual(s1, s2)
```

Comment:
This is pretty much the same as using `reflect.DeepEqual` instead of `cmp.Diff` like the original code. Not sure if we really want to use `cmp.Diff` here then.

Reply:
Or can't we utilize `status.Equal` here?

Reply:
> This is pretty much the same as using reflect.DeepEqual instead of cmp.Diff like the original code. Not sure if we really want to use cmp.Diff here then.

I agree with your opinion. 

We can't status.Equal, because status.Equal calls go-cmp as well.
cmp.Diff returns following diff.

```code
=== RUN   TestPreFilterDisabled
    node_ports_test.go:179: diff =   (*framework.Status)(
        - 	&⟪0x014000271580⟫{
        - 		code: 1,
        - 		err: &⟪0x014000528720⟫"fmt".wrapError{
        - 			msg: `reading "PreFilterNodePorts" from cycleState: not found`,
        - 			err: &⟪0x0140005415f0⟫"errors".errorString{s: "not found"},
        - 		},
        - 	},
        + 	&⟪0x014000271540⟫{
        + 		code: 1,
        + 		err: &⟪0x014000528700⟫"fmt".wrapError{
        + 			msg: `reading "PreFilterNodePorts" from cycleState: not found`,
        + 			err: &⟪0x010294a390⟫"errors".errorString{s: "not found"},
        + 		},
        + 	},
          )
```

I think the changes should be reverted. The problem may be caused by the data structure of *framework.Status.

Reply:
Other than this location, I would change to cmp.Diff due to the high readability of the errors.

Reply:
How about replacing `errors.New("not found")` in L178 with [`framework. ErrNotFound`](https://github.com/kubernetes/kubernetes/blob/7b243cef1a81f4a0ac1619f7b9238fcd2bac2b9c/pkg/scheduler/framework/cycle_state.go#L28)
(guessing go-cmp regards them as different because the root error object is different one. 🤔  )

Does it make the same result?

Reply:
I replaced `errors.New("not found") in L178 with `framework.ErrNotFound`, but it makes the same result.

```go
func TestPreFilterDisabled(t *testing.T) {
	pod := &v1.Pod{}
	nodeInfo := framework.NewNodeInfo()
	node := v1.Node{}
	nodeInfo.SetNode(&node)
	p, _ := New(nil, nil)
	cycleState := framework.NewCycleState()
	gotStatus := p.(framework.FilterPlugin).Filter(context.Background(), cycleState, pod, nodeInfo)
	wantStatus := framework.AsStatus(fmt.Errorf(`reading "PreFilterNodePorts" from cycleState: %w`, framework.ErrNotFound))
	if diff := cmp.Diff(*wantStatus, *gotStatus, cmp.AllowUnexported(framework.Status{}), cmpopts.EquateErrors()); diff != "" {
		t.Errorf("filter: status does not match (-want,+got):\n%s", diff)
		return
	}
}
```

```go
=== RUN   TestPreFilterDisabled
    node_ports_test.go:179: filter: status does not match (-want,+got):
          framework.Status{
          	code:    s"Error",
          	reasons: nil,
        - 	err: &⟪0x01400049f760⟫"fmt".wrapError{
        - 		msg: `reading "PreFilterNodePorts" from cycleState: not found`,
        - 		err: &⟪0x0102562390⟫"errors".errorString{s: "not found"},
        - 	},
        + 	err: &⟪0x01400049f740⟫"fmt".wrapError{
        + 		msg: `reading "PreFilterNodePorts" from cycleState: not found`,
        + 		err: &⟪0x0102562390⟫"errors".errorString{s: "not found"},
        + 	},
          	failedPlugin: "",
          }
```

A address of `err` field is different.

Reply:
Ah, got it. That's because now both of error's root error is the same ("not found"), but both of them are wrapped. H-m 🤔  Probably, we need to give up using the `cmp.Diff` here then. 

I can think of two options here:
- use `reflect.DeepEqual` as it was.
- create two assertions. One is `gotStatus.Code() == framework.Error`, the another is `gotStatus.Message() == "reading "PreFilterNodePorts" from ...`.

What do you think

Reply:
Ah, got it. That's because now both of error's root error is the same ("not found"), but both of them are wrapped. H-m 🤔 Probably, we need to give up using the cmp.Diff here then.

I can think of two options here:

use reflect.DeepEqual as it was.
create two assertions. One is gotStatus.Code() == framework.Error, the another is gotStatus.Message() == "reading "PreFilterNodePorts" from ....

Reply:
Yes, I am thinking the same thing. How about wrapping the reflect.DeepEqual with a cmp.Comparer so that we can see the difference as it is implemented now?
I know it is excessive as mentioned above, but it would make it easier to debug when this test fails in the future.

Reply:
doesn't make sense to me. 
If we use `reflect.DeepEqual` to compare and show diff from cmp.Diff, it's exactly the same as showing the whole status like it was (`t.Errorf("status does not match: %v, want: %v", gotStatus, test.wantStatus)`), isn't it.

Reply:
I mean:

```
=== RUN   TestPreFilterDisabled
    node_ports_test.go:179: diff =   (*framework.Status)(
        - 	&⟪0x014000271580⟫{
        - 		code: 1,
        - 		err: &⟪0x014000528720⟫"fmt".wrapError{
        - 			msg: `reading "PreFilterNodePorts" from cycleState: not found`,
        - 			err: &⟪0x0140005415f0⟫"errors".errorString{s: "not found"},
        - 		},
        - 	},
        + 	&⟪0x014000271540⟫{
        + 		code: 1,
        + 		err: &⟪0x014000528700⟫"fmt".wrapError{
        + 			msg: `reading "PreFilterNodePorts" from cycleState: not found`,
        + 			err: &⟪0x010294a390⟫"errors".errorString{s: "not found"},
        + 		},
        + 	},
          )
```

is not more valuable than the error log from `t.Errorf("status does not match: %v, want: %v", gotStatus, test.wantStatus)`

Reply:
OK, I use reflect.DeepEqual as it was.